tap_lightcast/streams.py

```diff
@@ -0,0 +1,109 @@
+"""Stream class for tap-lightcast."""
+
+import logging
+import requests
+from typing import Dict, Optional, Any
+from singer_sdk import typing as th
+from singer_sdk.streams import RESTStream
+from singer_sdk.authenticators import APIAuthenticatorBase, OAuthAuthenticator
+
+logging.basicConfig(level=logging.INFO)
+
+url_base = f"https://emsiservices.com/skills"
+auth_endpoint = f"https://auth.emsicloud.com/connect/token"
+
+
+class LightcastOAuthAuthenticator(OAuthAuthenticator):
+    """Authenticate with client credentials"""
+
+    @property
+    def oauth_request_body(self) -> dict:
+        return {
+            "client_id": self.config["client_id"],
+            "client_secret": self.config["client_secret"],
+            "grant_type": "client_credentials",
+            "scope": "emsi_open",
+        }
+
+
+class TapLightcastStream(RESTStream):
+    """Generic Lightcast stream class."""
+
+    @property
+    def url_base(self) -> str:
+        """Base URL of source"""
+        return url_base
+
+    @property
+    def authenticator(self) -> APIAuthenticatorBase:
+        return LightcastOAuthAuthenticator(
+            stream=self,
+            auth_endpoint=auth_endpoint,
+        )
+
+
+class SkillsList(TapLightcastStream):
+
+    client_id = self.config["client_id"]
```

Comment:
Same thing here.  If the values are not getting pulled in correctly, make sure you are using:

```
client_id = self.config.get("client_id")
client_secret = self.config.get("client_secret")
```